### PR TITLE
Exporter windowless

### PR DESCRIPTION
**Pull request by jburel**
Allow to save the images w/o passing all parameters
if no file specified,  a temporary file using the name of the image selected is created. We may want to change that and stop if no output specified.
To test: 
 * Open Fiji, update the b-f_plugins jar and all b-f jars if you are not already using develop jar (jars/bioformats)
 * Run the following code (new>script language>python)

```
from loci.plugins.in import ImporterOptions
from loci.plugins import BF
from ij import IJ
options = ImporterOptions()
options.setId(fileToView)
imps = BF.openImagePlus(options)
for imp in imps:
   imp.show()
   image = IJ.runPlugIn("loci.plugins.LociExporter", "outfile=fileToSaveTo.ome.tif windowless=true ")
```
Remove the outfile param to create the tmp file.

### CODE REVIEW

**Comment by pwalczysko** [comment]
Works as expected. Not sure if the comment about the ``metadata-extractor`` from https://github.com/openmicroscopy/bioformats/pull/1618#issuecomment-75968229 is valid here as well - I had the jar in my Fiji for this testing, so it certainly works WITH it.

Tested images were 1 ome.tiff and 1 lsm file. In both cases the new image was saved as expected.

**Comment by melissalinkert** [comment]
My only worry is that if ```outfile``` is not specified, the exported file path isn't going to be immediately obvious to the end user.  Creating a temp file is certainly fine, but if the path can be logged or returned somehow that might be nice.

**Comment by jburel** [comment]
@melissalinkert: record the name using the recorder. Other option will be to stop the process if no output file specified

File: components/bio-formats-plugins/src/loci/plugins/out/Exporter.java [hunk 1]
```diff
@@ -153,15 +176,36 @@ public void run() {
...
             }
         }
-
+        //create a temporary file if window less
+        if (windowless && (outfile == null || outfile.length() == 0)) {
+            try {
+                String name = removeExtension(imp.getTitle());
+                String n = name+ ".ome.tif";
+                File tmp = File.createTempFile(name, ".ome.tif");
+                File p = tmp.getParentFile();
```
Comment (melissalinkert) [inline components/bio-formats-plugins/src/loci/plugins/out/Exporter.java]: This results in an extra empty file being present in /tmp - could this either be deleted upon exit, or replaced with ```System.getProperty("java.io.tmpdir")```?

**Comment by melissalinkert** [comment]
Export works fine barring the minor nitpick above, but I'm still having trouble finding the logged file path if ```outfile``` is not set.  Even with the macro recorder running, I don't see the path being recorded anywhere.  Would using ```IJ.log``` instead of the macro recorder be an option?

Both are minor issues though, so I'm happy to merge if you would rather address them separately.

File: components/bio-formats-plugins/src/loci/plugins/out/Exporter.java (hunk 1 above)
Comment (jburel) [inline components/bio-formats-plugins/src/loci/plugins/out/Exporter.java]: oops forgot to add delete. It was in the initial code.

**Comment by jburel** [comment]
Will add path to log, might be easier.
I will do that in this PR

**Comment by jburel** [comment]
@melissalinkert: done

**Comment by melissalinkert** [comment]
Thanks, all looks great now.

**Comment by jburel** [comment]
--no-rebase

**Pull request merged**